### PR TITLE
Move signature to Authorization header.

### PR DESCRIPTION
**Pull request by woomurf**
### Issue
- [400 Error (Your client has issued a malformed or illegal request) ](https://www.notion.so/comcom/400-Error-Your-client-has-issued-a-malformed-or-illegal-request-bb514f745af34ae2a1c3cc0130e010cd)

### Description
- Move signature to Authorization header.

### Discussion Points
1. When use GET or DELETE method, Are params also included in querystring? for signature?
    1. Not use params, only use querystring.
    2. Add params in querystring. (But will have a duplicated information)

### CODE REVIEW

File: src/asset.ts [hunk 1]
```diff
@@ -21,14 +21,25 @@ export default class Asset {
     ethAddress: string,
     contractAddress?: string,
   ) {
-    const data = { appId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const querystring = { appId };
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/asset.ts]: Done! Changed to query!

File: src/asset.ts [hunk 2]
```diff
@@ -37,11 +48,23 @@ export default class Asset {
   }
 
   async getUserCreditBalance(appId: string, symbol: string, userId: string) {
-    const data = { appId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const querystring = { appId };
+    const trailingUrl = `credit/${appId}/${symbol}/${userId}`;
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/asset.ts]: Done!

File: src/auth.ts [hunk 3]
```diff
@@ -17,15 +17,25 @@ export default class Auth {
...
-    const data = {
+    const timestamp = Date.now();
+    const querystring = {
       appId,
       userId,
-      timestamp: Date.now(),
+    };
+    const data = {
+      method: 'GET',
+      path: '/auth/user',
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/auth.ts]: Done!

File: src/auth.ts [hunk 4]
```diff
@@ -34,32 +44,53 @@ export default class Auth {
...
+    const data = {
+      method: 'POST',
+      path: '/auth/user/ethAddress',
+      timestamp,
+      body: stringify(body),
     };
     const signature = this.ain.wallet.sign(stringify(data));
     return axios
-      .post(`${this.baseUrl}/user/ethAddress`, { data, signature })
+      .post(`${this.baseUrl}/user/ethAddress`, body, {
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/auth.ts]: Done!

File: src/event.ts [hunk 5]
```diff
@@ -37,12 +42,22 @@ export default class Event {
       startAt,
       endAt,
       platform,
-      timestamp: Date.now(),
+    };
+    const data = {
+      method: 'POST',
+      path: '/event/create',
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/event.ts]: Done!

File: src/discord.ts [hunk 6]
```diff
@@ -21,22 +21,45 @@ export default class Discord {
     discordServerId: string,
     userId: string
   ) {
-    const data = { appId, discordServerId, userId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const body = { appId, discordServerId, userId };
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/discord.ts]: Done!

File: src/auth.ts [hunk 7]
```diff
@@ -34,32 +44,53 @@ export default class Auth {
...
         throw e.response.data;
       });
   }
 
   removeUserEthAddress(appId: string, userId: string, ethAddress: string): Promise<User> {
-    const data = {
+    const timestamp = Date.now();
+    const querystring = {
       appId,
       userId,
```
[the remark this answers is not in the document]
Comment (woomurf) [reply src/auth.ts]: Done!

File: src/asset.ts [hunk 8]
```diff
@@ -21,14 +21,25 @@ export default class Asset {
     ethAddress: string,
     contractAddress?: string,
   ) {
-    const data = { appId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const query = { appId };
+    const trailingUrl = contractAddress
+      ? `nft/${chainId}/${ethAddress}/${contractAddress}`
+      : `nft/${chainId}/${ethAddress}`;
+    const data = {
```
Comment (platfowner) [inline src/asset.ts:29]: would it be useful to add a util function like `buildQueryData(method: string, path: string, timestamp: ...)`?

File: src/asset.ts [hunk 9]
```diff
@@ -21,14 +21,25 @@ export default class Asset {
...
+      method: 'GET',
+      path: `/asset/${trailingUrl}`,
+      timestamp,
+      querystring: stringify(query),
+    };
     const signature = this.ain.wallet.sign(stringify(data));
```
Comment (platfowner) [inline src/asset.ts:35]: would it be useful to add a util function like `signQueryData(data: any)` for readability even though it's a one-line function?

**Review by platfowner — APPROVED** [review]
LGTM with some comments.
Thanks!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!! 👍

File: src/asset.ts (hunk 8 above)
Comment (woomurf) [reply src/asset.ts]: Done! Add `buildData` util function!

File: src/asset.ts (hunk 9 above)
Comment (woomurf) [reply src/asset.ts]: Done! Add `signData` function in each class.

**Comment by woomurf** [comment]
Thanks for review! merging now~

**Pull request merged**